liboptv/src/multimed.c

```diff
@@ -0,0 +1,526 @@
+            beta2[i] = asin (sin(beta1) * mm->n1/mm->n2[i]);
+        beta3 = asin (sin(beta1) * mm->n1/mm->n3);
+
+        rbeta = (cal->ext_par.z0 - mm->d[0]) * tan(beta1) - Z * tan(beta3);
+        for (i = 0; i < mm->nlay; i++)
+            rbeta += (mm->d[i] * tan(beta2[i]));
+        
+        rdiff = r - rbeta;
+        rq += rdiff;
+        it++;
+    }
+    while (((rdiff > 0.001) || (rdiff < -0.001))  &&  it < n_iter);
+
+ 
+    if (it >= n_iter) {
+        printf ("multimed_r_nlay stopped after %d iterations\n", n_iter);  
+        return (1.0);
+    }
+
+    if (r != 0){ 
+        return (rq/r);
+    }  
+    else {
+        return (1.0);
+    }
+}
+
+
+/*  trans_Cam_Point() creates the shifted points for each position X,Y,Z 
+    Using Exterior and Interior parameters and the Glass vector of the variable
+    window position and the two vectors that point to the crossing point
+    
+    Arguments: 
+*/ 
+void trans_Cam_Point(Exterior ex
+                   , mm_np mm
+                   , Glass gl
+                   , vec3d pos
+                   , Exterior *ex_t
+                   , vec3d pos_t
+                   , double cross_p[3]
+                   , double cross_c[3]){
+
+/* Beat Luethi June 07: I change the stuff to a system perpendicular to the interface */
+    double dist_cam_glas,dist_point_glas,dist_o_glas; //glas inside at water 
+    int row, col;
+    double X,Y,Z;
+    
+    X = pos[0]; Y = pos[1]; Z = pos[2];
+    
+
+    dist_o_glas = sqrt( gl.vec_x * gl.vec_x + gl.vec_y * gl.vec_y + gl.vec_z * gl.vec_z);
+
+
+    dist_cam_glas = ex.x0 * gl.vec_x / dist_o_glas + ex.y0 * gl.vec_y / dist_o_glas + \
+    ex.z0 * gl.vec_z / dist_o_glas - dist_o_glas - mm.d[0];
+
+    dist_point_glas = X * gl.vec_x / dist_o_glas + \
+                    Y * gl.vec_y / dist_o_glas + \
+                    Z * gl.vec_z / dist_o_glas - dist_o_glas; 
+
+    cross_c[0] = ex.x0 - dist_cam_glas * gl.vec_x / dist_o_glas;
+    cross_c[1] = ex.y0 - dist_cam_glas * gl.vec_y / dist_o_glas;
+    cross_c[2] = ex.z0 - dist_cam_glas * gl.vec_z / dist_o_glas;
+
+    cross_p[0] = X - dist_point_glas * gl.vec_x / dist_o_glas;
+    cross_p[1] = Y - dist_point_glas * gl.vec_y / dist_o_glas;
+    cross_p[2] = Z - dist_point_glas * gl.vec_z / dist_o_glas;
+
+
+    for (row = 0; row < 3; row++)
+        for (col = 0; col < 3; col++)
+             ex_t->dm[row][col] = ex.dm[row][col];
+
+    ex_t->omega = ex.omega;
+    ex_t->phi   = ex.phi;
+    ex_t->kappa = ex.kappa;
+
+    ex_t->x0 = 0.;
+    ex_t->y0 = 0.;
+    ex_t->z0 = dist_cam_glas + mm.d[0];
+
+    pos_t[0]=sqrt( pow(cross_p[0] - (cross_c[0] - mm.d[0] * gl.vec_x / dist_o_glas ) ,2.)
+            +pow(cross_p[1] - (cross_c[1] - mm.d[0] * gl.vec_y / dist_o_glas ), 2.)
+            +pow(cross_p[2] - (cross_c[2] - mm.d[0] * gl.vec_z / dist_o_glas ), 2.));
+    pos_t[1] = 0;
+    pos_t[2] = dist_point_glas;
+}
+
+/* the opposite direction transfer from X_t,Y_t,Z_t to the X,Y,Z in 3D space */

```

Comment:
Finish documentation here too, and check other functions that they're in our standards.